### DISCUSSION TITLE
Paired Data X and Ys

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/fda-model#34] [Q&A]
When writing tests I found myself often confused when creating rating curves, flow freq curves, stage damage, etc out of paired data, because it's important which variable gets mapped to X, and which gets mapped to Y. 

For instance, if we try to integrate a damage-frequency relationship where damage is on the X-axis, our integration becomes invalid. 

Another case would be in composing functions. it matters whether a rating curve is stage-flow vs flow-stage. Mix them up, your curve composition is now garbage.

What options do we have for dealing with this/should we deal with it at all? A couple things I was pondering:

- We could rely on xml comments on the constructor to lay out how they these things ought to be assigned
- we could make separate classes to define each of our different relationships. Name the X and Y arrays the appropriate variable.

**Comment by HenryGeorgist** [comment]
I think that there are two basic approaches:
1. We use paired data, set it up right one time and be basically done with it.
2. We use the basic classes I stubbed out to accept arguments that make meaning, and set up the values to go into paired data in a clear way.

In go-consequences I took the approach of 1, in consequences (rust) I took the approach of 2. There is less boilerplate and objects in the second path, but the code is a bit harder to read unless we are careful in comments.
